src/App.tsx

```diff
@@ -17,18 +66,19 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter filter={filter} setFilter={setFilter} searchQuery={searchQuery} setSearchQuery={setSearchQuery}/>
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading ? <Loader /> : <TodoList todos={filteredTodos} onSelect={handleSelectTodo} />}
```

Comment:
Ensure that the `TodoList` component correctly handles the `todos` and `onSelect` props. Verify that these props are implemented in the `TodoList` component.